### PR TITLE
CC-18: New component for MultiSelector with tags

### PR DESCRIPTION
**Pull request by ankitp-atharva**
https://github.com/Components-care/backlog/issues/18
- We will reuse this component for CC-19 (https://github.com/Components-care/backlog/issues/19)

### CODE REVIEW

File: src/standalone/Selector/MultiSelectWithTags.tsx [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+
+	const handleGroupSelect = useCallback(
+		(selectedGroup: SelectorData | null) => {
+			if (onGroupSelect) onGroupSelect(selectedGroup);
+			if (selectedGroup !== null) {
+				const records = filteredData.filter(
+					(option: Data) =>
+						option.type?.toLowerCase() === selectedGroup.value.toLowerCase()
+				);
+				let filteredOptions = filteredData;
```
Comment (TQjona) [inline src/standalone/Selector/MultiSelectWithTags.tsx:126]:
```ts
const recordValues = records.map(record => record.value);
const filteredOptions = filteredData.filter(entry => !recordValues.includes(entry.value))
```

File: src/standalone/Selector/MultiSelectWithTags.tsx [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+		onGroupSelect,
+		selectedGroup,
+		enableIcons,
+		disable,
+		classes,
+	} = props;
+	const [selectedValues, setSelectedValue] = useState<Data[]>([]);
+	const input = useRef<TextFieldProps>();
+	const array = [...selected, ...selectedValues].flat();
+	const allSelected: Data[] = [];
```
Comment (TQjona) [inline src/standalone/Selector/MultiSelectWithTags.tsx:109]: you may need to disable the eslint rule which prevents usage of the non-null assert (`!`)
```ts
const allSelected = uniqueArray(array.map(item => item.value))
                                  .map(value => array.find(e => e.value === value)!)
```
uniqueArray is a utility function from SCF, you can copy paste it into the src/utils folder (don't forget to add documentation).

File: src/standalone/Selector/MultiSelectWithTags.tsx [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+				onSelect={handleGroupSelect}
+				onLoad={groupSelectLoadHandler}
+				clearable={true}
+				disable={disable}
+				multiSelect={false}
+			/>
+			<Autocomplete
+				key={filteredData.length}
+				freeSolo
+				id="cc-search-input"
```
Comment (TQjona) [inline src/standalone/Selector/MultiSelectWithTags.tsx:243]: we don't set HTML element ids

File: src/standalone/Selector/MultiSelectWithTags.tsx [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+				setSelectedValue(selectedValues);
+				if (input.current) input.current.value = "";
+			}
+		},
+		[setData, selectedValues, onSelect, selected, filteredData]
+	);
+
+	return (
+		<div>
+			<h3>{title}</h3>
```
Comment (TQjona) [inline src/standalone/Selector/MultiSelectWithTags.tsx:231]: use Material-UI Typography

File: src/standalone/Selector/MultiSelectWithTags.tsx [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+	/**
+	 * The currently selected values
+	 */
+	selected: Data[];
+	/**
+	 * The filtered data after selected dropdown value
+	 */
+	filteredData: Data[];
+	/**
+	 * Set filtered data after selected from search input
```
Comment (TQjona) [inline src/standalone/Selector/MultiSelectWithTags.tsx:56]: Clearify please, is this some kind of onChange event for `filteredData`? If yes please include it in the comment

**Review by TQjona — CHANGES_REQUESTED** [review]

File: src/standalone/Selector/MultiSelectWithTags.tsx (hunk 2 above)
Comment (ankitp-atharva) [reply src/standalone/Selector/MultiSelectWithTags.tsx]: Util files added as arrayFunctions.

File: src/standalone/Selector/MultiSelectWithTags.tsx (hunk 5 above)
Comment (ankitp-atharva) [reply src/standalone/Selector/MultiSelectWithTags.tsx]: yes, it's callback function so comment updated

File: src/utils/arrayFunctions.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Gets the uniq elements from giver array
+ * @param array The array object
+ */
+export const uniqueArray = <T>(array: T[]): T[] => {
```
Comment (TQjona) [inline src/utils/arrayFunctions.ts:5]: function isn't exported correctly + file name should match function name

**Review by TQjona — CHANGES_REQUESTED** [review]

File: src/utils/arrayFunctions.ts (hunk 6 above)
Comment (ankitp-atharva) [reply src/utils/arrayFunctions.ts]: @TQjona ,  file name updated and fixed export default

**Review by TQjona — APPROVED** [review]

**Pull request merged**